templates/repo/actions/list.tmpl

```diff
@@ -76,6 +76,11 @@
 						</button>
 					{{end}}
 				</div>
+
+				{{if .AllowTriggerWorkflowDispatchEvent}}
```

Comment:
```suggestion
				{{if .WorkflowDispatchConfig}}
```
`AllowTriggerWorkflowDispatchEvent` is unnecessary

Reply:
`AllowTriggerWorkflowDispatchEvent` is used to decide whether to display prompts and buttons, and `WorkflowDispatchConfig` is used to decide whether to process inputs; 
it can also be triggered without inputs.

Reply:
When workflow_dispatch is defined, but inputs are not
→ workflowDispatchConfig = {}
→ workflowDispatchConfig != nil, let's display the notification
→ user click `Run Workflow`
→ workflowDispatchConfig.Inputs = nil, don't have inputs, don't show them

When workflow_dispatch and inputs are both defined
→ workflowDispatchConfig = nil
→ don't display the notification

But we should do some changes to `WorkflowDispatchConfig` function.
`WorkflowDispatchConfig` will return `nil` when workflow_dispatch is defined, but inputs are not. 
And this also causes a bug now, when workflow_dispatch is defined, but inputs are not, the notification will not display😕

Reply:
Yes, yes, you are right; before, WorkflowDispatchConfig() always returned non-null, but now the behavior of parse has been modified. Use workflowDispatchConfig to determine whether to display the button or not.

Reply:
Sorry for such a confusing scenario; I fixed it and the logic now is:
* workflow_dispatch is `not` defined, WorkflowDispatchConfig() return `nil`
* workflow_dispatch is defined WorkflowDispatchConfig() return `non nil` , Whether the `input` is defined or not
* inputs is not defined `WorkflowDispatch.Inputs` is nil

so,
* `WorkflowDispatchConfig` if exists, display button
*  When `.WorkflowDispatchConfig.Inputs` is nil, `<range >` nothing happens

Reply:
`ParseRawOn` is no longer needed
but, I think some modif still need to be made, workflow_dispatch is not defined, WorkflowDispatchConfig() return nil

```
var config WorkflowDispatch
node := val["workflow_dispatch"]
if !decodeNode(node, &config) {
    return nil
}

return &config
```

change to
```
var config WorkflowDispatch
node, found := val["workflow_dispatch"]
if found && decodeNode(node, &config) {
    return &config
}

return nil
```

Reply:
Yes. Thanks for your contribution.

ps: not only `workflow_dispatch`, other events,  e.g. `workflow_call`, have some similar functions,
we can check them later.

Reply:
Okay, I think I will have some time to work on it next few days; btw, it seems that the `On` function has the same problem, switch does not handle all types of `kind`.

Reply:
I pushed a new commit to https://gitea.com/gitea/act/pulls/85 and added a recursive function to handle nested SequenceNode. It can also support workflow_call. The following cases have been tested.

```yaml
on: workflow_dispatch
on: [push, workflow_dispatch]
on:
    - push
    - workflow_dispatch
on:
    - push
    - [pull_request, workflow_dispatch]
on:
    - push:
    - pull_request
    - [push, pull_request]
    - workflow_dispatch:
        inputs:
            logLevel:
                description: 'Log level'
                required: true
                default: 'warning'
                type: choice
                options:
                - info
                - warning
                - debug
```